test/integration/test_create_dss_docker_image.py

```diff
@@ -167,3 +190,104 @@ def test_docker_socket_on_host_touched(dss_container_context, fake_docker_socket
         wait_for_socket_access(container)
 
     assert stat_before == socket.stat()
+
+
+def assert_exec_run(container: Container, command: str, **kwargs) -> str:
+    """
+    Execute command in container and verify success.
+    """
+    exit_code, output = container.exec_run(command, **kwargs)
+    output = output.decode("utf-8").strip()
+    assert exit_code == 0, output
+    return output
+
+
+class SocketInspector:
+    def __init__(self, context_provider, socket_on_host: Path):
+        self._context_provider = context_provider
+        self.socket_on_host = socket_on_host
+        self._container = None
+
+    @contextmanager
+    def context(self):
+        with self._context_provider(self.socket_on_host) as container:
+            wait_for_socket_access(container)
+            self._container = container
```

Comment:
this unusual, usually you would return an object, that encapsulates the container and all the functions you want to run on it. or you make the class itself the context manager via __enter__, __exit__

Reply:
See latest push for an updated version